### PR TITLE
Pod disruption budget v1 beta1 This resolves #8290

### LINKED ISSUE #8290: Remove support for PodDisruptionBudget v1beta1
**Note: This should be implemented only after the 0.35.0 release which drops support for Kubernetes 1.19 & 1.20!**

Today, Strimzi supports PodDisruptionBudgets v1beta1 and v1. PodDisruptionBudget v1 has been introduced in Kubernetes 1.21. So once we drop support for Kube 1.19 and 1.20, we can also remove support for PDB v1beta1.

### PR DESCRIPTION
**Pull request by ShazaAldawamneh**
### Type of change

_Select the type of your PR_

- Refactoring

### Description
 This resolves #8290
Searching the code base for v1beta1.PodDisruptionBudget and going through the places where it is used, from where the methods are called etc and  remove support for PDB v1beta1.

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java [hunk 1]
```diff
@@ -1534,9 +1534,7 @@ public PodDisruptionBudget generatePodDisruptionBudget() {
      *
      * @return The PodDisruptionBudget V1Beta1.
      */
-    public io.fabric8.kubernetes.api.model.policy.v1beta1.PodDisruptionBudget generatePodDisruptionBudgetV1Beta1() {
-        return PodDisruptionBudgetUtils.createCustomControllerPodDisruptionBudgetV1Beta1(componentName, namespace, labels, ownerReference, templatePodDisruptionBudget, replicas);
-    }
+
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java:1537]: No need for the double empty line.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/PodDisruptionBudgetUtils.java [hunk 2]
```diff
@@ -124,43 +91,5 @@ public static PodDisruptionBudget createCustomControllerPodDisruptionBudget(
...
-                    .withAnnotations(TemplateUtils.annotations(template))
-                    .withOwnerReferences(ownerReference)
-                .endMetadata()
-                .withNewSpec()
-                    .withMinAvailable(new IntOrString(Math.max(0, replicas - (template != null ? template.getMaxUnavailable() : DEFAULT_MAX_UNAVAILABLE))))
-                    .withSelector(new LabelSelectorBuilder().withMatchLabels(labels.strimziSelectorLabels().toMap()).build())
-                .endSpec()
-                .build();
-    }
+
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/PodDisruptionBudgetUtils.java:94]: No need for the double empty line -> you probably don't need any empty line at the end.

File: operator-common/src/main/java/io/strimzi/operator/common/operator/resource/PodDisruptionBudgetV1Beta1Operator.java [hunk 3]
```diff
@@ -10,23 +10,3 @@
...
-     */
-    public PodDisruptionBudgetV1Beta1Operator(Vertx vertx, KubernetesClient client) {
-        super(vertx, client, "v1beta1.PodDisruptionBudget");
-    }
-
-    @Override
-    protected MixedOperation<PodDisruptionBudget, PodDisruptionBudgetList, Resource<PodDisruptionBudget>> operation() {
-        return client.policy().v1beta1().podDisruptionBudget();
-    }
-}
```
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/common/operator/resource/PodDisruptionBudgetV1Beta1Operator.java:32]: Just delete the whole class.

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/PodDisruptionBudgetV1Beta1OperatorTest.java [hunk 4]
```diff
@@ -17,53 +17,3 @@
 import static java.util.Collections.singletonMap;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
-
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/PodDisruptionBudgetV1Beta1OperatorTest.java:20]: Just delete the whole class.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java [hunk 5]
```diff
@@ -682,15 +678,7 @@ protected Future<Void> podDisruptionBudget() {
      *
      * @return  Completes when the PDB was successfully created or updated
      */
-    protected Future<Void> podDisruptionBudgetV1Beta1() {
-        if (pfa.hasPodDisruptionBudgetV1()) {
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java:686]: The `pfa.hasPodDisruptionBudgetV1()` should not be needed anymore. So it should be removed from the `PlatformFeaturesAvailability` class as well. You should also update the `podDisruptionBudget()` method above to not use it either.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java [hunk 6]
```diff
@@ -782,20 +782,6 @@ public PodDisruptionBudget generatePodDisruptionBudget(boolean stableIdentities)
         }
     }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:784]: No need for the double empty lines here which happen after you delete the method.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java [hunk 7]
```diff
@@ -486,14 +486,6 @@ public PodDisruptionBudget generatePodDisruptionBudget() {
         return PodDisruptionBudgetUtils.createPodDisruptionBudget(componentName, namespace, labels, ownerReference, templatePodDisruptionBudget);
     }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java:488]: No need for the double empty lines here which happen after you delete the method.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java [hunk 8]
```diff
@@ -454,14 +454,6 @@ public PodDisruptionBudget generatePodDisruptionBudget() {
         return PodDisruptionBudgetUtils.createPodDisruptionBudget(componentName, namespace, labels, ownerReference, templatePodDisruptionBudget);
     }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java:456]: No need for the double empty lines here which happen after you delete the method.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/PodDisruptionBudgetUtils.java [hunk 9]
```diff
@@ -51,39 +51,6 @@ public static PodDisruptionBudget createPodDisruptionBudget(
                 .build();
     }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/PodDisruptionBudgetUtils.java:53]: No need for the double empty lines here which happen after you delete the method.

**Review by scholzj — COMMENTED** [review]
Nice work. Looks mostly good. I left some comments around the formatting and some mores tuff to delete.

**Review by scholzj — COMMENTED** [review]
Nice work. Looks mostly good. I left some comments around the formatting and some mores tuff to delete.

**Comment by scholzj** [comment]
You will also need to fix the DCO-signoff -> the instructions should be under the _Details_ link under the DCO status. In general, when committing, you should add the `-s` option ... e.g. `git commit -s -m "My commit message"`

**Comment by ShazaAldawamneh** [comment]
This resolves #8290

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java [hunk 10]
```diff
@@ -782,20 +782,6 @@ public PodDisruptionBudget generatePodDisruptionBudget(boolean stableIdentities)
...
-     *
-     * @return The pod disruption budget V1Beta1.
-     */
-    public io.fabric8.kubernetes.api.model.policy.v1beta1.PodDisruptionBudget generatePodDisruptionBudgetV1Beta1(boolean stableIdentities) {
-        if (stableIdentities) {
-            return PodDisruptionBudgetUtils.createCustomControllerPodDisruptionBudgetV1Beta1(componentName, namespace, labels, ownerReference, templatePodDisruptionBudget, replicas);
-        } else {
-            return PodDisruptionBudgetUtils.createPodDisruptionBudgetV1Beta1(componentName, namespace, labels, ownerReference, templatePodDisruptionBudget);
-        }
-    }
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java:785]: The double emoty line is not needed probably.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java [hunk 11]
```diff
@@ -471,14 +467,8 @@ protected Future<Void> podDisruptionBudget() {
...
      */
-    protected Future<Void> podDisruptionBudgetV1Beta1() {
-        if (pfa.hasPodDisruptionBudgetV1()) {
-            return Future.succeededFuture();
-        } else {
-            return podDisruptionBudgetV1Beta1Operator
-                    .reconcile(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperStatefulSetName(reconciliation.name()), zk.generatePodDisruptionBudgetV1Beta1())
-                    .map((Void) null);
-        }
+    protected Future<Void> podDisruptionBudgetV1() {
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java:470]: This seems wrong => this does not do anything and should be removed. The v1 version of the PDB is managed by the method right above.

**Review by scholzj — APPROVED** [review]
Thanks for the PR

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks for the PR!

**Comment by scholzj** [comment]
Thanks for the PR.

**Pull request merged**